### PR TITLE
Fixes up pooling  and dropout related onnx import issues

### PR DESCRIPTION
**Pull request by agibsonccc**
## What changes were proposed in this pull request?
Fixes up drop out import by mapping ratio optionally as both an input
or as an attribute via a import pre hook.

Also fixes up minor padding issue with max and average pooling.

Adds back onnx converter usign the java cpp presets. This is established to work on linux and doesn't necessarily need to support windows to be useful.

(Please fill in changes proposed in this fix)

## How was this patch tested?

(Please explain how this patch was tested. E.g. unit tests, integration tests, manual tests)

## Quick checklist

The following checklist helps ensure your PR is complete:

- [X ] Eclipse Contributor Agreement signed, and signed commits - see [IP Requirements](https://deeplearning4j.org/eclipse-contributors) page for details
- [ X] Reviewed the [Contributing Guidelines](https://github.com/eclipse/deeplearning4j/blob/master/CONTRIBUTING.md) and followed the steps within.
- [X ] Created tests for any significant new code additions.
- [ X] Relevant tests for your changes are passing.

### CODE REVIEW

File: nd4j/samediff-import/samediff-import-api/src/main/kotlin/org/nd4j/samediff/frameworkimport/hooks/PreImportHook.kt [hunk 1]
```diff
@@ -50,7 +50,6 @@ interface PreImportHook {
         importGraph: ImportGraph<GeneratedMessageV3, GeneratedMessageV3, GeneratedMessageV3, GeneratedMessageV3, GeneratedMessageV3, GeneratedMessageV3, ProtocolMessageEnum>
     ): HookResult {
         // Parameter docs below are from the onnx operator docs:
-        // https://github.com/onnx/onnx/blob/master/docs/Operators.md#slice
```
Comment (treo) [inline nd4j/samediff-import/samediff-import-api/src/main/kotlin/org/nd4j/samediff/frameworkimport/hooks/PreImportHook.kt:53]: Was that link removed intentionally?

File: platform-tests/pom.xml [hunk 2]
```diff
@@ -49,8 +43,10 @@
 
     <properties>
         <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+        <maven.compiler.source>1.7</maven.compiler.source>
+        <maven.compiler.target>1.7</maven.compiler.target>
         <dl4j.version>1.0.0-SNAPSHOT</dl4j.version>
-        <platform.classifier>${os.name}-${os.arch}</platform.classifier>
+        <platform.classifier>windows-x86_64</platform.classifier>
```
Comment (treo) [inline platform-tests/pom.xml]: Maybe have a profile for this? In principle that should pick up the correct value anyway, so I'm not sure why you need to change it

**Review by treo — CHANGES_REQUESTED** [review]

File: nd4j/samediff-import/samediff-import-api/src/main/kotlin/org/nd4j/samediff/frameworkimport/hooks/PreImportHook.kt (hunk 1 above)
Comment (agibsonccc) [reply nd4j/samediff-import/samediff-import-api/src/main/kotlin/org/nd4j/samediff/frameworkimport/hooks/PreImportHook.kt:53]: Yes the hook doesn't actually have anything to do with the split op. It was just left over.

File: platform-tests/pom.xml (hunk 2 above)
Comment (agibsonccc) [reply platform-tests/pom.xml]: On windows it picks up something like: windows-AMD64 or something off. I'm not sure why. In practice I usually override this when running CI builds. I get what you mean though.

File: nd4j/samediff-import/samediff-import-api/src/main/kotlin/org/nd4j/samediff/frameworkimport/hooks/PreImportHook.kt (hunk 1 above)
Comment (treo) [reply nd4j/samediff-import/samediff-import-api/src/main/kotlin/org/nd4j/samediff/frameworkimport/hooks/PreImportHook.kt:53]: Then you should probably remove the line above it too.

**Review by treo — APPROVED** [review]

**Pull request merged**